File: src/compiler/typescript_to_rust/generators/ic_object/functions/msg_cycles_refunded128.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionMsgCyclesRefunded128(): Rust {
    return /* rust */ `
        fn _azle_ic_msg_cycles_refunded128(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            let return_value: boa_engine::bigint::JsBigInt = ic_cdk::api::call::msg_cycles_refunded128().into();
            Ok(return_value.into())
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate_rearchitecture/src/ic/msg_cycles_refunded128.rs
use proc_macro2::TokenStream;
use quote::quote;

pub fn generate() -> TokenStream {
    quote! {
        fn msg_cycles_refunded128<'a>(
            context: &'a JSContextRef,
            _this: &CallbackArg,
            _args: &[CallbackArg],
        ) -> Result<JSValueRef<'a>, anyhow::Error> {
            let return_js_value: JSValue =
                candid::encode_one(ic_cdk::api::call::msg_cycles_refunded128())?.into();
            to_qjs_value(&context, &return_js_value)
        }
    }
}
